aws/aws.go

```diff
@@ -422,14 +423,29 @@ func GetAllResources(targetRegions []string, excludeAfter time.Time, resourceTyp
 
 				resourcesCache["S3"] = make(map[string][]*string)
 
-				for bucketRegion, bucketName := range bucketNamesPerRegion {
-					resourcesCache["S3"][bucketRegion] = bucketName
+				for bucketRegion, bucketNames := range bucketNamesPerRegion {
+          var matchedBuckets []*string
+
+          for _, re := range configObj.S3FilterRule.IncludeNamesRE {
+            for _, bucketName := range aws.StringValueSlice(bucketNames) {
+              if re.MatchString(bucketName) {
+                matchedBuckets = append(matchedBuckets, aws.String(bucketName))
+              }
+            }
+          }
+
+          if len(configObj.S3FilterRule.IncludeNamesRE) > 0 {
+            resourcesCache["S3"][bucketRegion] = matchedBuckets
+          } else {
+            resourcesCache["S3"][bucketRegion] = bucketNamesPerRegion[bucketRegion]
+          }
```

Comment:
I think it would be cleaner to have this logic embedded in `getAllS3Buckets`, considering that there is already some filtering happening there (the logic with `excludeAfter` timestamp)

Reply:
Are you thinking it would make sense within `getBucketInfo` [here](https://github.com/rhoboat/cloud-nuke/blob/matching-s3-108/aws/s3.go#L208)?

Reply:
Ah yup that is a better place!